unstructured/ingest/connector/airtable.py

```diff
@@ -49,39 +60,117 @@ class AirtableIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
     """
 
     config: SimpleAirtableConfig
-    file_meta: AirtableFileMeta
+    table_meta: AirtableTableMeta
     registry_name: str = "airtable"
 
     @property
     def filename(self):
         return (
             Path(self.standard_config.download_dir)
-            / self.file_meta.base_id
-            / f"{self.file_meta.table_id}.csv"
+            / self.table_meta.base_id
+            / f"{self.table_meta.table_id}.csv"
         ).resolve()
 
     @property
     def _output_filename(self):
         """Create output file path based on output directory, base id, and table id"""
-        output_file = f"{self.file_meta.table_id}.json"
-        return Path(self.standard_config.output_dir) / self.file_meta.base_id / output_file
+        output_file = f"{self.table_meta.table_id}.json"
+        return Path(self.standard_config.output_dir) / self.table_meta.base_id / output_file
+
+    @property
+    def date_created(self) -> Optional[str]:
+        return self.file_metadata.date_created
+
+    @property
+    def date_modified(self) -> Optional[str]:
+        return self.file_metadata.date_modified
+
+    @property
+    def exists(self) -> Optional[bool]:
+        return self.file_metadata.exists
+
+    @property
+    def record_locator(self) -> Optional[Dict[str, Any]]:
+        return {
+            "base_id": self.table_meta.base_id,
+            "table_id": self.table_meta.table_id,
+            "view_id": self.table_meta.view_id,
+        }
+
+    @property
+    def source_url(self) -> Optional[str]:
+        return self.file_metadata.source_url
+
+    @requires_dependencies(["pyairtable"], extras="airtable")
+    def _get_table_rows(self):
+        from pyairtable import Api
+
+        api = Api(self.config.personal_access_token)
+        try:
+            table = api.table(self.table_meta.base_id, self.table_meta.table_id)
+            table_url = table.url
+            rows = table.all(
+                view=self.table_meta.view_id,
+            )
+        except Exception as e:
+            # TODO: more specific error handling?
+            logger.error(e)
+            return None, None
+
+        if len(rows) == 0:
+            logger.info("Empty document, retrieved table but it has no rows.")
+        return rows, table_url
+
+    @cached_property
+    def file_metadata(self):
+        """Gets file metadata from the current table."""
+        rows, table_url = self._get_table_rows()
```

Comment:
Not sure how `@cached_property` or if the underlying variable used to cache it available outside this method, but since you're using `self._get_table_rows()` to derive the values, would be nice for the cached value to be set in that method if it's called before this one so that we don't have to make that call twice. 

Also general question, values such as `date_created` shouldn't be changing, but `date_modified` might, so I'm not sure if we would want to avoid caching that value. @ryannikolaidis, not how the intention of this property might impact how we get this value.

Reply:
@rbiseck3 [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property) only caches the property for the lifetime of that IngestDoc instance. I don't think we need to be concerned about changes that occur during the duration of processing a document, with respect to these properties.

Reply:
ideally we don't want separate get calls for each piece of data here since in most cases a single `get` is going to give us most or all of this and it's probably unlikely that they'll change over the span of processing a document and should be able to query the value at any time, but to your point (@rbiseck3 ) it seems like the most technically accurate way to do this would be to explicitly set (or overwrite) the mutable values _at the time_ of fetching the doc to insure that we write the correct metadata of the raw document when fetched.

Reply:
any thoughts?